src/Sulu/Bundle/ContentBundle/Repository/ResourceLocatorRepository.php

```diff
@@ -155,4 +174,22 @@ private function implodeRlpParts(StructureInterface $structure, array $parts, $s
 
         return $title;
     }
+
+    /**
+     * Returns the first ancestor-or-self of the given document which is not a ghost-document.
+     * If all ancestor documents are ghost-documents, the root of the documents is returned.
+     *
+     * @param object    $document
+     *
+     * @return object
+     */
+    private function getNonGhostAncestor($document)
+    {
+        while ($this->documentInspector->getLocalizationState($document) === LocalizationState::GHOST &&
+        $document instanceof ParentBehavior) {
```

Comment:
Please indent correctly:

```php
while ($this->documentInspector->getLocalizationState($document) === LocalizationState::GHOST
    && $document instanceof ParentBehavior
) {
    // ...
}
```

The `&&` should be in front, because it is important to know how the two statements are connected.